pkg/controller/job/job_controller.go

```diff
@@ -1757,3 +1768,17 @@ func countReadyPods(pods []*v1.Pod) int32 {
 	}
 	return cnt
 }
+
+// This checks if we should apply PodRecreationPolicy.
+// PodRecreationPolicy controls when we recreate pods if they are marked as terminating
+// Failed means that we recreate only once the pod has terminated.
+func hasPodReplacementPolicyFailed(job *batch.Job) bool {
+	return feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) && *job.Spec.PodReplacementPolicy == batch.Failed
```

Comment:
or hasPodFailurePolicy

Reply:
We default in cases of PodFailurePolicy so I figured just checking for Failed is sufficient.  

We also only allow Failed as an option in validation.

Reply:
But the behavior of PodFailurePolicy should change even if JobPodReplacementPolicy is disabled. That is why we did an update to KEP 3329.

A behavioral change is permitted for a beta feature. That's one of the reasons why we go through a beta period: to learn from our mistakes and improve semantics.

Reply:
I pushed an update for this.  Its a bit awkward to handle `Failed` without counting terminating so I had to add some logic for case when `PodFailurePolicy` is enabled but `PodRecreationPolicy` is disabled.  I count terminating but do not update the status field.